src/node_file.cc

```diff
@@ -1321,8 +1333,15 @@ static void Rename(const FunctionCallbackInfo<Value>& args) {
 
   BufferValue old_path(isolate, args[0]);
   CHECK_NOT_NULL(*old_path);
+  THROW_IF_INSUFFICIENT_PERMISSIONS(
+      env, permission::PermissionScope::kFileSystemIn, *old_path);
+  THROW_IF_INSUFFICIENT_PERMISSIONS(
+      env, permission::PermissionScope::kFileSystemOut, *old_path);
```

Comment:
Why do we need write access to the old path?

Reply:
Generally speaking, the `rename` call basically deletes the old path. So, writing access must be required.

`mv old_path new_path` -> old_path will not exist